app/code/Magento/Newsletter/Model/Subscriber.php

```diff
@@ -350,6 +350,19 @@ public function loadByEmail($subscriberEmail)
         return $this;
     }
 
+    /**
+     * Load subscriber data from resource model by email ansd storeId
+     *
+     * @param string $subscriberEmail
+     * @param int $storeId
+     * @return $this
+     */
+    public function loadByEmailAndStore($subscriberEmail, $storeId)
```

Comment:
Unfortunately this class marked as `@api`, so according to https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/ we can't new public methods into it